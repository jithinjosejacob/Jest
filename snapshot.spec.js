var itemStock =[
 {'Id':'1', 'ItemName':'Razors', 'Stock':'10' },
 {'Id':'2', 'ItemName':'Socks', 'Stock':'1'  },
 {'Id':'3', 'ItemName':'Lungi', 'Stock':'5' },
 {'Id':'4', 'ItemName':'Socks', 'Stock':'2'  },
];

function filterItemStock(arr, key, term){
    return arr.filter(function(obj){
        return obj[key] === term
    });
}

describe('SnapShot Testing',()=>{

    it('Nonsnapshot - Return all item with matching id',()=>{
        expect (filterItemStock(itemStock, 'Id' ,'1')).toEqual([
            {'Id':'1', 'ItemName':'Razors', 'Stock':'10' }
        ])
    })
    it('Nonsnapshot - all items with matching Name',()=>{
        expect (filterItemStock(itemStock, 'ItemName' ,'Socks')).toEqual([
            {'Id':'2', 'ItemName':'Socks', 'Stock':'1' },
            {'Id':'4', 'ItemName':'Socks', 'Stock':'2' }
        ])
    })

    it('Snapshot - Return all item with matching id',()=>{
        expect (filterItemStock(itemStock, 'Id' ,'1')).toMatchSnapshot()
    })
    
    it('Snapshot - Return all items with matching Name',()=>{
        expect (filterItemStock(itemStock, 'ItemName' ,'Socks')).toMatchSnapshot()
    })

})
