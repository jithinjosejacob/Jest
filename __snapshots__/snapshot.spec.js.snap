// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Testing Snapshot - Return all item with matching id 1`] = `
Array [
  Object {
    "Id": "1",
    "ItemName": "Razors",
    "Stock": "10",
  },
]
`;

exports[`SnapShot Testing Snapshot - Return all items with matching Name 1`] = `
Array [
  Object {
    "Id": "2",
    "ItemName": "Socks",
    "Stock": "1",
  },
  Object {
    "Id": "4",
    "ItemName": "Socks",
    "Stock": "2",
  },
]
`;
